eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -107,13 +147,12 @@ public function index(FullTextData $fullTextData)
             } else {
                 $integerValue = 0;
             }
-            // Split text on whitespace
-            $wordArray = explode(' ', $fullTextValue->value);
+            // Split transformed text on whitespace
+            $wordArray = explode(' ', $this->transformationProcessor->transform($fullTextValue->value, $this->transformationProcessorCommands));
```

Comment:
This change aligns with behavior of [`FullText` `CriterionHandler`](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FullText.php#L194).

It was uncovered when trying to [create Content item with phrase containing hyphens](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/158413582#L818). `TransformationProcessor` replaces hyphens with spaces, which results in treating hyphenated word as separate words.

Issues:
- Should hyphenated word be treated as separate words? (I guess it should - it works like this in Solr, but want to make sure).
- It would be better to somehow reuse [configuration array for `TransformationProcessor` in `FullText` `CriterionHandler`](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FullText.php#L31), but I don't know how to do it in a clean way.
- Let me know if this change should be in a separate PR.

Feedback ping @andrerom @pspanja